Execution of visualization of [Panic] failed

### Discord username

_No response_

### What type of issue is this?

Intermittent – Occurring irregularly

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

Sometimes when working with the projects the drop-downs are not showing up. Them message in the console says:
```router.ts:62 [ERROR] app/gui/src/controller/graph/widget.rs:153 Failed to modify widget visualization: Peer has replied with an error: Error { code: 2006, message: "Visualization not found", data: None }.```
Full log below
[localhost-1694789315003.log](https://github.com/enso-org/enso/files/12620644/localhost-1694789315003.log)
[enso-project-manager-2023-09-15.0.log](https://github.com/enso-org/enso/files/12620664/enso-project-manager-2023-09-15.0.log)


### Expected behaviour

The drop down should appear

### How we can reproduce it?

_No response_

### Screenshots or screencasts

_No response_

### Logs

_No response_

### Enso Version

nightly 15.09

### Browser or standalone distribution

Standalone distribution (local project)

### Browser Version or standalone distribution

standalone

### Operating System

MacOS

### Operating System Version

_No response_

### Hardware you are using

_No response_

I don't see any visualization timeouts in the log. There is a timeout with VCS initialization
```
[ERROR] [2023-09-15T12:35:07+02:00] [org.enso.languageserver.requesthandler.vcs.InitVcsHandler] Initialize project request [Number(2)] for [2c261afa-d735-4bfe-80f2-d7f001c1ee01] failed with: null.
java.util.concurrent.TimeoutException: null
```
but it seems that it should be addressed by the #7898

> I don't see any visualization timeouts in the log. There is a timeout with VCS initialization
> 
> ```
> [ERROR] [2023-09-15T12:35:07+02:00] [org.enso.languageserver.requesthandler.vcs.InitVcsHandler] Initialize project request [Number(2)] for [2c261afa-d735-4bfe-80f2-d7f001c1ee01] failed with: null.
> java.util.concurrent.TimeoutException: null
> ```
> 
> but it seems that it should be addressed by the #7898

Do you get that regularly? I don't think my PR will fix that.

Reproduced while testing cloud. Pasting screenshot and logs:
Also the problem is it freezed the whole electron instance, cant reload nor switch back to dashboard.

<img width="1728" alt="Screenshot 2023-10-13 at 13 53 05" src="https://github.com/enso-org/enso/assets/2855109/47103cab-75eb-42c1-920c-5cac23101553">


```
Starting Enso Runtime in version
Enso Compiler and Runtime
Version:    2023.2.1-nightly.2023.10.13.1
Built with: scala-2.13.11 for GraalVM 17.0.7
Built from: develop* @ a96f2d7aba141780116cc49cb501fada757a0ef6
Running on: OpenJDK 64-Bit Server VM, GraalVM Community, JDK 17.0.7+7-jvmci-23.0-b12
            Linux 5.4.17-2136.323.8.2.el8uek.x86_64 (amd64)

[INFO] [2023-10-13T11:33:53Z] [org.enso.languageserver.boot.LanguageServerComponent] Starting Language Server...
[INFO] [2023-10-13T11:33:53Z] [org.enso.languageserver.boot.MainModule] Initializing main module of the Language Server from [2023.2.1-nightly.2023.10.13.1, LanguageServerConfig(0.0.0.0,30001,None,30002,None,00000000-0000-0000-0000-000000000001,/volumes/workspace/project_root,ProfilingConfig(None,None,None),StartupConfig(false),language-server,scala.concurrent.impl.ExecutionContextImpl$$anon$3@4879dfad[Running, parallelism = 1, size = 1, active = 0, running = 0, steals = 1, tasks = 0, submissions = 0]), INFO]
[INFO] [2023-10-13T11:34:00Z] [org.enso.languageserver.runtime.RuntimeConnector] Starting the runtime connector.
[INFO] [2023-10-13T11:34:00Z] [org.enso.languageserver.runtime.RuntimeConnector] Runtime connector established connection with the message endpoint [com.oracle.truffle.api.instrumentation.TruffleInstrument$Env$MessageTransportProxy$MessageEndpointProxy@33e06a53].
[INFO] [2023-10-13T11:34:02Z] [org.enso.languageserver.search.SuggestionsHandler] Starting suggestions handler from [Config(ContentRootWithFile(Project(00000000-0000-0000-0000-000000000001),/volumes/workspace/project_root),FileManagerConfig(3 seconds,1),VcsManagerConfig(5 seconds,3 seconds,true),PathWatcherConfig(5 seconds,5 seconds,10),ExecutionContextConfig(5 seconds),ProjectDirectoriesConfig(/volumes/workspace/project_root),ProfilingConfig(None,None,None),StartupConfig(false),None), org.enso.searcher.sql.SqlSuggestionsRepo@50d15982].
[INFO] [2023-10-13T11:34:02Z] [org.enso.languageserver.boot.MainModule] Main module of the Language Server initialized with config [Config(ContentRootWithFile(Project(00000000-0000-0000-0000-000000000001),/volumes/workspace/project_root),FileManagerConfig(3 seconds,1),VcsManagerConfig(5 seconds,3 seconds,true),PathWatcherConfig(5 seconds,5 seconds,10),ExecutionContextConfig(5 seconds),ProjectDirectoriesConfig(/volumes/workspace/project_root),ProfilingConfig(None,None,None),StartupConfig(false),None)].
[INFO] [2023-10-13T11:34:04Z] [org.enso.languageserver.boot.LanguageServerComponent] Started server at json:0.0.0.030001, , binary:0.0.0.0:30002
[INFO] [2023-10-13T11:34:42Z] [org.enso.languageserver.protocol.json.JsonConnectionController] Initializing resources.
[INFO] [2023-10-13T11:34:42Z] [org.enso.languageserver.boot.resource.DirectoriesInitialization] Initializing directories...
[INFO] [2023-10-13T11:34:42Z] [org.enso.languageserver.boot.resource.DirectoriesInitialization] Initialized directories.
[INFO] [2023-10-13T11:34:42Z] [org.enso.languageserver.boot.resource.JsonRpcInitializationComponent] Initializing JSON-RPC protocol.
[INFO] [2023-10-13T11:34:44Z] [org.enso.languageserver.boot.resource.JsonRpcInitializationComponent] JSON-RPC protocol initialized.
[INFO] [2023-10-13T11:34:44Z] [org.enso.languageserver.boot.resource.ZioRuntimeInitialization] Initializing ZIO runtime...
[INFO] [2023-10-13T11:34:45Z] [org.enso.languageserver.boot.resource.ZioRuntimeInitialization] ZIO runtime initialized [org.enso.languageserver.effect.ExecutionContextRuntime@1d7d77e4].
[INFO] [2023-10-13T11:34:45Z] [org.enso.languageserver.boot.resource.RepoInitialization] Initializing sql database [org.enso.searcher.sql.SqlDatabase@17242300]...
[INFO] [2023-10-13T11:34:46Z] [org.enso.languageserver.boot.resource.RepoInitialization] Initialized sql database [org.enso.searcher.sql.SqlDatabase@17242300].
[INFO] [2023-10-13T11:34:46Z] [org.enso.languageserver.boot.resource.TruffleContextInitialization] Initializing Runtime context...
[INFO] [2023-10-13T11:34:55Z] [org.enso.languageserver.boot.resource.TruffleContextInitialization] Initialized Runtime context.
[INFO] [2023-10-13T11:34:55Z] [org.enso.languageserver.boot.resource.RepoInitialization] Initializing suggestions repo [/volumes/workspace/project_root/.enso/suggestions.db]...
[INFO] [2023-10-13T11:34:55Z] [org.enso.languageserver.filemanager.ReceivesTreeUpdatesHandler] Initialized
[INFO] [2023-10-13T11:34:55Z] [org.enso.languageserver.search.SuggestionsHandler] Initializing: Truffle context initialized.
[INFO] [2023-10-13T11:34:56Z] [org.enso.languageserver.search.SuggestionsHandler] Initializing: project name is updated to [Project_root].
[INFO] [2023-10-13T11:34:56Z] [org.enso.languageserver.filemanager.PathWatcher] Initialized [class org.enso.filewatcher.WatcherAdapterFactory] for [Path(00000000-0000-0000-0000-000000000001,Vector())].
[INFO] [2023-10-13T11:34:56Z] [org.enso.languageserver.protocol.binary.BinaryConnectionController] Connection closed [89.64.61.30].
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.search.SuggestionsHandler] Initializing: got type graph response.
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.RepoInitialization] Initialized Suggestions repo [/volumes/workspace/project_root/.enso/suggestions.db].
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.search.SuggestionsHandler] Initializing: suggestions repo initialized.
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.protocol.json.JsonConnectionController] RPC session initialized for client [7f7ecc1c-91a4-4aae-add9-35c3fec95363].
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.protocol.json.JsonConnectionController] Json session terminated [7f7ecc1c-91a4-4aae-add9-35c3fec95363].
[INFO] [2023-10-13T11:34:57Z] [akka.actor.LocalActorRef] Message [org.enso.jsonrpc.ResponseResult] from Actor[akka://language-server/user/json-connection-controller-0b924116-7642-47ae-9607-7657936fbb44#-2042150551] to Actor[akka://language-server/user/message-handler-supervisor-241ef695-d4e8-492d-bc35-9ac07576a4da/message-handler-0b924116-7642-47ae-9607-7657936fbb44#-949634667] was not delivered. [1] dead letters encountered, no more dead letters will be logged in next [5.000 min]. If this is not an expected behavior then Actor[akka://language-server/user/message-handler-supervisor-241ef695-d4e8-492d-bc35-9ac07576a4da/message-handler-0b924116-7642-47ae-9607-7657936fbb44#-949634667] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.protocol.json.JsonConnectionController] Initializing resources.
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.DirectoriesInitialization] Initializing directories...
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.DirectoriesInitialization] Initialized directories.
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.JsonRpcInitializationComponent] Initializing JSON-RPC protocol.
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.JsonRpcInitializationComponent] JSON-RPC protocol initialized.
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.ZioRuntimeInitialization] Initializing ZIO runtime...
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.ZioRuntimeInitialization] ZIO runtime initialized [org.enso.languageserver.effect.ExecutionContextRuntime@1d7d77e4].
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.RepoInitialization] Initializing sql database [org.enso.searcher.sql.SqlDatabase@17242300]...
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.RepoInitialization] Initialized sql database [org.enso.searcher.sql.SqlDatabase@17242300].
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.TruffleContextInitialization] Initializing Runtime context...
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.TruffleContextInitialization] Initialized Runtime context.
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.RepoInitialization] Initializing suggestions repo [/volumes/workspace/project_root/.enso/suggestions.db]...
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.boot.resource.RepoInitialization] Initialized Suggestions repo [/volumes/workspace/project_root/.enso/suggestions.db].
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.protocol.json.JsonConnectionController] RPC session initialized for client [30559f5c-b382-4d78-b9ae-8d53c1dcf185].
[INFO] [2023-10-13T11:34:57Z] [org.enso.languageserver.protocol.binary.BinaryConnectionController] Data session initialized for client: 30559f5c-b382-4d78-b9ae-8d53c1dcf185 [89.64.61.30].
[INFO] [2023-10-13T11:34:58Z] [enso] Background jobs started.
[INFO] [2023-10-13T11:34:59Z] [org.enso.languageserver.text.CollaborativeBuffer] Buffer opened for [path:Path(00000000-0000-0000-0000-000000000001,Vector(src, Main.enso)), client:30559f5c-b382-4d78-b9ae-8d53c1dcf185].
[WARN] [2023-10-13T11:35:02Z] [enso.org.enso.librarymanager.local.DefaultLocalLibraryProvider] Local library search path [***/libraries] does not exist.
[INFO] [2023-10-13T11:35:02Z] [enso.org.enso.interpreter.runtime.DefaultPackageRepository] Found library Standard.Base @ 2023.2.1-nightly.2023.10.13.1 at [***/2023.2.1-nightly.2023.10.13.1].
[INFO] [2023-10-13T11:35:03Z] [enso.org.enso.interpreter.runtime.DefaultPackageRepository] Found library Standard.Table @ 2023.2.1-nightly.2023.10.13.1 at [***/2023.2.1-nightly.2023.10.13.1].
[INFO] [2023-10-13T11:35:03Z] [enso.org.enso.interpreter.runtime.DefaultPackageRepository] Found library Standard.Database @ 2023.2.1-nightly.2023.10.13.1 at [***/2023.2.1-nightly.2023.10.13.1].
[INFO] [2023-10-13T11:35:04Z] [enso.org.enso.interpreter.runtime.DefaultPackageRepository] Found library Standard.AWS @ 2023.2.1-nightly.2023.10.13.1 at [***/2023.2.1-nightly.2023.10.13.1].
[INFO] [2023-10-13T11:35:26Z] [enso] Background jobs started.
[INFO] [2023-10-13T11:35:35Z] [org.enso.languageserver.text.CollaborativeBuffer] Buffer opened for [path:Path(00000000-0000-0000-0000-000000000001,Vector(src, Main.enso)), client:30559f5c-b382-4d78-b9ae-8d53c1dcf185].
[INFO] [2023-10-13T11:35:35Z] [enso.org.enso.interpreter.runtime.DefaultPackageRepository] Found library Standard.Visualization @ 2023.2.1-nightly.2023.10.13.1 at [***/2023.2.1-nightly.2023.10.13.1].
[ERROR] [2023-10-13T11:35:44Z] [org.enso.languageserver.requesthandler.search.CompletionHandler] Request [Number(12)] timed out.
[ERROR] [2023-10-13T11:36:21Z] [org.enso.languageserver.search.SuggestionsHandler] Cannot Remove [Local(Some(7314394e-98dc-45b2-86db-b57b1ccb41e1),local.Project_root.Main,operator1,Standard.Base.Any.Any,Scope(Position(6,6),Position(7,54)),None)].
[ERROR] [2023-10-13T11:36:29Z] [org.enso.languageserver.search.SuggestionsHandler] Cannot Remove [Function(Some(6db2ae1b-329d-4999-9ef1-6246c55b6875),local.Project_root.Main,operator2,List(Argument(<internal-30>,Standard.Base.Any.Any,false,false,None,None)),Standard.Base.Any.Any,Scope(Position(6,7),Position(8,24)),None)].
[WARN] [2023-10-13T11:43:07Z] [enso] Execution of visualization [VisualizationConfiguration(4ba46b09-c02a-456a-978b-c8e24ccf6374,ModuleMethod(MethodPointer(Standard.Visualization.Widgets,Standard.Visualization.Widgets,get_widget_json),Vector(.from_keys_and_values, ['keys', 'values', 'error_on_duplicates'])),local.Project_root.Main)] on value [78df9442-13ee-41d5-b127-ff979a0f7d92] of [Panic] failed.
[ERROR] [2023-10-13T11:43:14Z] [org.enso.languageserver.search.SuggestionsHandler] Cannot Add [Local(Some(ae75c2ab-7e00-4dea-b573-e032909c75b2),local.Project_root.Main,map1,Standard.Base.Any.Any,Scope(Position(6,7),Position(10,14)),None)].
[ERROR] [2023-10-13T11:45:05Z] [org.enso.languageserver.search.SuggestionsHandler] Cannot Add [Local(Some(cc04fcb8-fe0f-470d-9c49-3e2175e23ba4),local.Project_root.Main,operator4,Standard.Base.Any.Any,Scope(Position(6,7),Position(10,29)),None)].
[ERROR] [2023-10-13T11:45:06Z] [org.enso.languageserver.search.SuggestionsHandler] Cannot Remove [Function(Some(cc04fcb8-fe0f-470d-9c49-3e2175e23ba4),local.Project_root.Main,operator4,List(Argument(<internal-763>,Standard.Base.Any.Any,false,false,None,None)),Standard.Base.Any.Any,Scope(Position(6,7),Position(10,21)),None)].
[ERROR] [2023-10-13T11:46:25Z] [org.enso.languageserver.search.SuggestionsHandler] Cannot Add [Local(Some(0c9b3cab-5657-46aa-adcd-e244644c8aa3),local.Project_root.Main,operator7,Standard.Base.Any.Any,Scope(Position(6,7),Position(13,31)),None)]
```

--- a/engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala
+++ b/engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala
@@ -17,7 +17,7 @@ import org.enso.interpreter.instrument.profiling.ExecutionTime
 import org.enso.interpreter.instrument._
 import org.enso.interpreter.node.callable.FunctionCallInstrumentationNode.FunctionCall
 import org.enso.interpreter.node.expression.builtin.meta.TypeOfNode
-import org.enso.interpreter.runtime.`type`.Types
+import org.enso.interpreter.runtime.`type`.{Types, TypesGen}
 import org.enso.interpreter.runtime.callable.function.Function
 import org.enso.interpreter.runtime.control.ThreadInterruptedException
 import org.enso.interpreter.runtime.error.{
@@ -484,17 +484,19 @@ object ProgramExecutionSupport {
       case Left(error) =>
         val message =
           Option(error.getMessage).getOrElse(error.getClass.getSimpleName)
-        ctx.executionService.getLogger.log(
-          Level.WARNING,
-          "Execution of visualization [{0}] on value [{1}] of [{2}] failed.",
-          Array[Object](
-            visualization.config,
-            expressionId,
-            Try(TypeOfNode.getUncached.execute(expressionValue))
-              .getOrElse(expressionValue.getClass),
-            error
+        val typeOfNode = Try(TypeOfNode.getUncached.execute(expressionValue))
+        if (!typeOfNode.map(TypesGen.isPanicSentinel).getOrElse(false)) {
+          ctx.executionService.getLogger.log(
+            Level.WARNING,
+            "Execution of visualization [{0}] on value [{1}] of [{2}] failed.",
+            Array[Object](
+              visualization.config,
+              expressionId,
+              typeOfNode.getOrElse(expressionValue.getClass),
+              error
+            )
           )
-        )
+        }
         ctx.endpoint.sendToClient(
           Api.Response(
             Api.VisualizationEvaluationFailed(

